### PR TITLE
feat: integrate docs project(WIP)

### PR DESCRIPTION
**Pull request by proke03**
## Description
I add the storybook tab on charts.org like below image. And I integrated it into this repository.
It's been maintained with different lint setups so far, so it's hard to fully integrate them all at once.

## Related Issue
#3268

## Screenshots (if appropriate):
image
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by nikolasrieble — COMMENTED** [review]
Why not just call the folder `storybook` instead of `org`?

File: org/docs/183.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
...se 3:if(!(c=p.sent())||e.isDisposed())return[2];for(l=[],f=0,d=c;f<d.length;f++)h=d[f],l.push({resource:o.Sf.parse(h.fileName),edit:{range:this._textSpanToRange(e,h.textSpan),text:n}});return[2,{edits:l}]}}))}))},t}(g);function N(e){P=W(e,"typescript")}function E(e){I=W(e,"javascript")}function K(){return new Promise((function(e,t){if(!I)return t("JavaScript not registered!");e(I)}))}function R(){return new Promise((function(e,t){if(!P)return t("TypeScript not registered!");e(P)}))}function W(e,t){var n=new u(t,e),r=function(){for(var e=[],t=0;t<arguments.length;t++)e[t]=arguments[t];return n.getLanguageServiceWorker.apply(n,e)},i=new b(r);return o.Mj.registerCompletionItemProvider(t,new y(r)),o.Mj.registerSignatureHelpProvider(t,new S(r)),o.Mj.registerHoverProvider(t,new w(r)),o.Mj.registerDocumentHighlightProvider(t,new k(r)),o.Mj.registerDefinitionProvider(t,new x(i,r)),o.Mj.registerReferenceProvider(t,new C(i,r)),o.Mj.registerDocumentSymbolProvider(t,new M(r)),o.Mj.registerDocumentRangeFormattingEditProvider(t,new D(r)),o.Mj.registerOnTypeFormattingEditProvider(t,new A(r)),o.Mj.registerCodeActionProvider(t,new O(r)),o.Mj.registerRenameProvider(t,new T(r)),new v(i,e,t,r),r}}}]);
```
Comment (nikolasrieble) [inline org/docs/183.js:1]: These are generated files, right? Why commit them? Maybe a git ignore update is required first.

Comment (proke03) [reply org/docs/183.js:1]: charts.org also includes files in docs.
I think this is used for deployment using github pages.

**Comment by proke03** [comment]
I totally agree with you.
But I thought I couldn't do it all at once.

**Comment by proke03** [comment]
I think both are possible. And I also think the latter is better. 
And the storybook is shown as an iframe in this PR.

**Comment by ckifer** [comment]
First of all thank you @proke03 for the POC and work here!

I think I am fine with this, but maybe we should take smaller steps at it. We don't want two versions of the docs that are out of sync with each other, that is no fun.

How about maybe lets take a step back, integrate the storybook itself here first, set ourselves up as a monorepo (recharts/recharts-docs or recharts/org doesn't matter to me), and then we can move the docs site over into the same sub-repo the storybook is in?

**Comment by ckifer** [comment]
Hmm... 

This is actually interesting. We can only host one GH pages site per repository. If we move the storybook here and host it we can't host the docs site here and if we host the docs site here we can't host the storybook. 

I can't seem to find any other way other than iframe to embed a storybook. Can you?

**Comment by proke03** [comment]
> We can only host one GH pages site per repository.

I didn't know that. I will try putting both in one folder.

**Comment by ckifer** [comment]
Even if you put them in one folder it doesn't solve the issue. In order for the iframe to work it has to be deployed separately no?

**Comment by proke03** [comment]
Since it is not going to be distributed separately, I think I need to find a way other than iframe.

**Comment by proke03** [comment]
In my opinion, it is better to minimize the use of tools. 
Ideally, only use github pages for org distribution.

**Comment by ckifer** [comment]
Agreed with @proke03 - minimize the tools minimize the things people need access to. 

If we deploy the storybook here we get the benefit of updating documentation in the same repo. When it deploys it will automatically be updated in the iframe on the org site. So you could go to it directly or via the org site and we wouldn't need to update the org site very often once the storybook is in a good spot.

**Comment by Yilun-Sun** [comment]
Agreed with you

**Comment by proke03** [comment]
I will continue to develop this PR.
However, as shown above, the history is too messy, so I am experimenting with github actions in other branch.

**Comment by ckifer** [comment]
Sounds good. Thank you!

**Pull request closed without merging**